pypeit/core/gui/skysub_regions.py

```diff
@@ -0,0 +1,552 @@
+                      horizontalalignment='center', verticalalignment='center')
+        if yesno:
+            self.axes['info'].fill_between([0.8, 0.9], 0, 1, facecolor='green', alpha=0.5, transform=self.axes['info'].transAxes)
+            self.axes['info'].fill_between([0.9, 1.0], 0, 1, facecolor='red', alpha=0.5, transform=self.axes['info'].transAxes)
+            self.axes['info'].text(0.85, 0.5, "YES", transform=self.axes['info'].transAxes,
+                          horizontalalignment='center', verticalalignment='center')
+            self.axes['info'].text(0.95, 0.5, "NO", transform=self.axes['info'].transAxes,
+                          horizontalalignment='center', verticalalignment='center')
+        self.axes['info'].set_xlim((0, 1))
+        self.axes['info'].set_ylim((0, 1))
+        self.canvas.draw()
+
+    def add_region(self):
+        """
+        Add/subtract a defined region
+        """
+        # Figure out the locations of the start values
+        ys = np.argmin(np.abs(self._start[1]-self._spectrace))
+        difs = self.tslits_dict['slit_righ'][ys, self._currslit] - self.tslits_dict['slit_left'][ys, self._currslit]
+        sval = (self._start[0]-self.tslits_dict['slit_left'][ys, self._currslit]) / difs
+        sidx = int(round(self._resolution*sval))
+        # Figure out the locations of the start values
+        yf = np.argmin(np.abs(self._end[1]-self._spectrace))
+        diff = self.tslits_dict['slit_righ'][yf, self._currslit] - self.tslits_dict['slit_left'][yf, self._currslit]
+        fval = (self._end[0]-self.tslits_dict['slit_left'][yf, self._currslit]) / diff
+        fidx = int(round(self._resolution*fval))
+        # Switch the indices if needed
+        if sidx > fidx:
+            sidx, fidx = fidx, sidx
+        # Check that we are within bounds
+        if sidx < 0:
+            sidx = 0
+        if fidx > self._resolution:
+            fidx = self._resolution
+        # Assign the sky regions
+        self._skyreg[self._currslit][sidx:fidx] = self._addsub
+        # If some regions are removed, remove this from the "all slits" regions, as well
+        if self._addsub == 0:
+            self._allreg[sidx:fidx] = 0
+
+    def add_region_all(self):
+        """ Set the sky regions for all slits simultaneously
+        """
+        # Do some checks
+        xmin, xmax = self._start[0], self._end[0]
+        if xmax < xmin:
+            xmin, xmax = xmax, xmin
+        if xmin < 0:
+            xmin = 0
+        if xmax > self._resolution:
+            xmax = self._resolution
+        # Apply to all slits
+        for sl in range(self._nslits):
+            self._skyreg[sl][xmin:xmax] = self._addsub
+        # Set the all regions parameter
+        self._allreg[xmin:xmax] = self._addsub
+
+    def reset_regions(self):
+        self._skyreg = [np.zeros(self._resolution, dtype=np.bool) for all in range(self._nslits)]
+        self._allreg[:] = False
+        return
+
+
```

Comment:
I think this should be @classmethod, rather than a stand-alone method, i.e. 

@classmethod
def initialize(cls, det, frame, tslits_dict, runtime=False, printout=False)

I won't insist on the American spelling of initialize;)

Reply:
A class method is a function you can run attached to the class, but that does not require an instance of the class. I think that is the appropriate mode for this function since it has no meaning outside of the class. 

Joe

Reply:
I've read a bit more about this, and you're exactly right - this is a perfect case of using an **@classmethod**. I've updated the code.